### ISSUE TITLE
[1.7.0.1] Uncaught Error: Cannot find module '/node_modules/meteor/socket-stream-client/sockjs-0.3.4.js'

### ISSUE THREAD

**Issue by holmrenser** [meteor/meteor#9928]
After updating to 1.7.0.1 everything runs fine in development, but after bundling I get the following error:

```
692f20a306f2c5e6a421b2984812cd77a74cc51f.js?meteor_js_resource=true:7 Uncaught Error: 
Cannot find module '/node_modules/meteor/socket-stream-client/sockjs-0.3.4.js'
```

This did not happen when using 1.6

**Comment by KoenLav** [comment]
Can you consistently reproduce this?

I have seen this error as well, but it disappeared after running it again and hasn't reappeared since.

**Comment by tjramage** [comment]
@KoenLav — This happens to me every time I try to load my app. I think it's a knock-on effect of another issue though (I'm wondering if dynamic-import is broken?). See console output below:

<img width="973" alt="screen shot 2018-06-02 at 22 11 34" src="https://user-images.githubusercontent.com/5517057/40880924-05b1fa46-66b2-11e8-8d8d-e88745be9024.png">

**Comment by LoriKarikari** [comment]
I'm having the same issue when I try to access the website on another computer within my local network. I can access it on localhost:3000 on my dev machine but when I try to access it on 192.168.0.7:3000 on another machine, I'm getting the same errors as @tjramage posted above. It worked before updating.

**Comment by LoriKarikari** [comment]
@ramijarrar thanks! It works now.
@holmrenser I haven't tested it on a server yet, but will report if it happens there too

**Comment by abdulrahman-khankan** [comment]
Have the same problem after upgrading from 1.6.1.1 to 1.7.0.1, only on production (with ROOT_URL set). Locally it works fine, even when using --production.

**Comment by jeffreyflynt** [comment]
I am having this issue as well in production.  I reverted the commits to wait on a fix.

**Comment by jon617** [comment]
Similar issue here.  Seems to be connected to the value (or absence of) environment variable ROOT_URL.  Error happened for me on a dev staging server after doing a `meteor update` and setting the `ROOT_URL=127.0.0.1:3000` (or whichever meteor port in use) fixed my problem for now, but makes me nervous about pushing to production.

**Comment by jeffreyflynt** [comment]
Update:

I was able to fix this by using the actual domain name in the root url instead of localhost in production.  I had it set in meteor settings, but I also needed it set in:

Environment=ROOT_URL=https://domain

when starting the service.

**Comment by spencern** [comment]
Saw this same issue when building for production but using a demo/localhost `ROOT_URL` in our docker-compose file to test production builds locally. Resolved by changing from
`ROOT_URL: "http://localhost"` => `ROOT_URL: "http://127.0.0.1"

**Comment by abdulrahman-khankan** [comment]
Changed ROOT_URL on production from localhost to the root hostname, and now it doesn't produce an error in the console. However, it doesn't show any data from the server. Just empty pages/components.
Weirdly though, it works fine for another colleague connected within the same network (I connect through a VPN -GlobalProtect- but it works fine when reverting to 1.6.1.1).

**Comment by LoriKarikari** [comment]
Yesterday everything was working fine on my local network with the ROOT_URL workaround, but today I'm suddenly getting the errors below when I access the app on my local network. The app was working when I stopped working on it yesterday but today I can't access it anymore. I don't know if it's related to this issue though.

![meteor-errors](https://user-images.githubusercontent.com/7902980/41041930-b3b00f34-69a0-11e8-82fb-085afa7138b6.PNG)

**Comment by jon617** [comment]
FYI to those looking to downgrade to meteor 1.6.1.2, here's what I did.

1. Made a backup copy of `.meteor/packages` from my code directory, so I have a list of installed meteor packages.

2. In a new/temporary directory, `meteor create test --bare --release 1.6.1.2`

3. That created a new `test` directory with version 1.6.1.2.  In the `.meteor/` directory created there, copied the following files into my code's `.meteor/` directory: `release`, `versions`, `platforms`, `packages`, and `.finished-upgraders`

4. Looked through the backup of `.meteor/packages` created in step 1, and did a `meteor add` on each package, for example: `meteor add session email accounts-base react-meteor-data`

**Comment by LoriKarikari** [comment]
@jon617 I downgraded with `meteor update --release 1.6.2`

**Comment by jon617** [comment]
Thanks @LoriKarikari that would have been a lot easier

**Comment by srimenow** [comment]
Had a similar error after upgrading to 1.7 and trying to test on android device, got to love Chrome remote debugger!

```
ROOT_URL="http://192.168.*.**:3000" meteor run
```

makes everything happy, thanks to everyone for giving their input to make this a relatively painless fix :)
Values ```localhost``` and ```127.0.0.1``` makes the websocket inaccessible over the network.

Decided to take the gamble and deploy into production.

**Comment by holmrenser** [comment]
I stumbled across a solution in the pull request for release 1.7.0.1 (#9917):

As per @benjamn's  [comment](https://github.com/meteor/meteor/pull/9917#issuecomment-395808847) apparently sockjs-0.3.4 is no longer automatically used. 

His fix worked for me (copied from the comment):

> 1. First, copy the socket-stream-client package from a checkout of this repository into your local packages/ directory.
> 2. Next, open socket-stream-client/browser.js in your text editor.
> 3. Add a static import "./sockjs-0.3.4.js" line to the top of that file, so that SockJS will be bundled in the initial JS bundle, and WebSockets will not be used.
> 4. Run/test/deploy your app as usual.

**Comment by benjamn** [comment]
I've just released a new version of `socket-stream-client` that goes back to using `SockJS` in the modern bundle (just like the legacy bundle) rather than a native `WebSocket`. If you run `meteor update socket-stream-client` to get version 0.2.2, I believe these problems will go away, since `SockJS` has all sorts of clever tricks for coping with unusual networking environments that can cause problems for plain `WebSocket`s.

**Comment by chamithkanchana** [comment]
> @LoriKarikari We had the same issue and were able to get it working on other machines by setting ROOT_URL explicitly (e.g `ROOT_URL="http://192.168.0.7:3000" meteor run`) - I think it may be related to the changes in #9439

@ramijarrar You saved my day bro! Thank you very much.
